### PR TITLE
Fixed #2082. Sub_6D31A6 now working.

### LINKED ISSUE #2082: Fix sub_6D31A6
https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/ride/ride.c#L4172

This implementation does not work because track_block_get_previous called from track_circuit_iterator_previous seems to be ending prematurely and not iterating the complete track. This can be reproduced by constructing a giga coaster with a lift hill (pre-designed one will do) and testing it. This means other methods that use track_block_get_previous could be faulty. It might a particular track block that causes it.

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Fixed #2082. Sub_6D31A6 now working.
Small mistake in track previous meant the x/y coordinates didn't match the map element. There were also a few other issues that presented themselves once that was fixed.

### CODE REVIEW

**Comment by IntelOrca** [comment]
Given you have made a PR for this I will assume you want code improvement feedback, so here we go.

File: src/ride/ride.c [hunk 1]
```diff
@@ -1188,11 +1188,14 @@ int sub_6C683D(int* x, int* y, int* z, int direction, int type, uint16 extra_par
...
 		}
+		
 		if (flags & (1 << 3)) {
+			// Cable lift hill track
 			mapElement->properties.track.colour |= (1 << 3);
 		}
 		if (flags & (1 << 4)) {
-			mapElement->properties.track.colour &= 0xF7;
+			// Clear Cable lift hill track
+			mapElement->properties.track.colour &= ~(1 << 3);
```
Comment (IntelOrca) [inline src/ride/ride.c]: Like I have done with lift hill, make the functions is_track_cable_lift_hill(trackElement) and set_track_cable_lift_hill(bool).

File: src/ride/ride.c [hunk 2]
```diff
@@ -4306,13 +4309,6 @@ bool ride_create_vehicles(rct_ride *ride, int rideIndex, rct_xy_element *element
  */
 static bool sub_6D31A6(rct_ride *ride, bool isApplying)
```
Comment (IntelOrca) [inline src/ride/ride.c]: I guess you can probably rename this to validate_cable_lift_hill or something. Its checking and setting track flags, so I guess its doing more than just validating. Perhaps try_initialise_cable_lift_hill or something.

**Comment by duncanspumpkin** [comment]
Okay I think that covers it. I'll merge this when travis/appveyor passes

**Pull request merged**